portal/src/components/Materials/Materials.component.js

```diff
@@ -54,10 +54,41 @@ export default {
       }
     },
     /**
-     * Select material
-     * @param material - {Object}
+     * Select material from collection
      */
     selectMaterial(material) {
+      if (this.selectFor === 'delete') {
+        this.deleteMaterial(material)
+      } else {
+        this.toggleMaterial(material)
+      }
+    },
+    /**
+     * Delete material from collection
+     * @param material - {Object}
+     */
+    deleteMaterial(material) {
+      const { id } = this.$route.params
+      this.$store
+        .dispatch('removeMaterialFromMyCollection', {
+          collection_id: id,
+          data: [{ external_id: material.external_id }]
+        })
+        .then(() => {
+          Promise.all([
+            this.$store.dispatch('getMaterialInMyCollection', {
+              id,
+              params: { page: 1, page_size: 10 }
+            }),
+            this.$store.dispatch('getCollection', id)
+          ]).then(() => null)
+        })
+    },
+    /**
+     * Toggle material from selection
+     * @param material - {Object}
+     */
+    toggleMaterial(material) {
```

Comment:
Misschien deze implementatie nog vervangen door wat ik suggereerde?

```
if (this.value.indexOf(material.external_id) === -1) {
    return [...this.value, material.external_id]
else {
    return this.value.filter(
        item => item !== material.external_id
    )
}
```

En dan in `selectMaterial`: `this.$emit('input', this.toggleMaterial(material))`.

Of ben je het daar niet mee eens?

Reply:
ik snapte niet dat je dat je bedoelde